### DISCUSSION TITLE
replace manually mapping process for GOLD biosample IDs

### DISCUSSION THREAD

**Discussion by aclum** [microbiomedata/issues#990] [General]
The current process to add GOLD biosample identifiers is to manually do the mapping and update existing biosample records with changesheets. This currently has to happen for samples which are ingested through the submission portal. Ideally this is handled programmatically. Could this be added to existing ELT code? Either submission portal to mongo or GOLD ETL?  
cc @mslarae13 @sujaypatil96 @pkalita-lbl

**Comment by pkalita-lbl** [comment]
Are you talking about setting the `gold_biosample_identifiers` slot on `Biosample` records? How do you manually determine what values to put in that slot?

**Comment by aclum** [comment]
Yes, i am talking about setting the gold_biosample_id for class Biosample records. Right now we either determine this information looking at the sample names and/or by input from the PI. I was/am advocating for this to be part a submission portal record so we don't have to track down this information after the fact, then the ETL code could populate this information. A similar issue came up today when Brandon put NCBI identifiers in the source_mat_id slot.

**Comment by pkalita-lbl** [comment]
Okay, importing the `gold_biosample_identifiers` into `submission-schema` classes would be a pretty easy thing to do if that's all that's needed.

**Comment by aclum** [comment]
Okay ill defer to @mslarae13 on when to schedule this relative to other submission portal priorities.
